### PR TITLE
Surgery to cut-off proctoring

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/NEX-711
https://oat-sa.atlassian.net/browse/NEX-792

### CODE REVIEW

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
lgtm

**Comment by mike-ionut-mihai-sandu-tao** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Comment by gabrielfs7** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: helper/DataTableHelper.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA ;
```
Comment (gabrielfs7) [inline helper/DataTableHelper.php]:
```suggestion
 * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
```

File: helper/DataTableHelper.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+ *
+ */
+
+namespace oat\taoTestCenterRostering\helper;
+
+use oat\taoClientDiagnostic\model\storage\PaginatedStorage;
+
+/**
+ * Provides common data helper for datatable component.
+ * @todo Move this class in taoCore, make an abstract/interface and its implementation
```
Comment (gabrielfs7) [inline helper/DataTableHelper.php]: Could you please add it here? https://oat-sa.atlassian.net/wiki/spaces/NEX/pages/364579446/Technical+Debt

File: helper/DataTableHelper.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+ */
+class DataTableHelper
+{
+    /**
+     * The default number of rows displayed in a page
+     */
+    const DEFAULT_ROWS = 25;
```
Comment (gabrielfs7) [inline helper/DataTableHelper.php]: - [ ] Please, add visibility for the constants

File: helper/DataTableHelper.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    /**
+     * Paginates a collection to render a subset in a table
+     * @param array|PaginatedStorage $collection - The full amount of lines to paginate
+     * @param array [$options] - Allow to setup the page. These options are supported:
+     * - self::OPTION_ROWS : The number of rows per page
+     * - self::OPTION_PAGE : The index of the page to get
+     * @param function [$dataRenderer] - An optional callback function provided to format the paginated data
+     * @return array
+     */
+    public static function paginate($collection, $options = array(), $dataRenderer = null)
```
Comment (gabrielfs7) [inline helper/DataTableHelper.php]: - [ ] Can we make this non static?
- [ ] Can we add proper type hint and return type?
- [ ] Cam we avoid double type parameters like $collection and have 2 different methods instead?

File: helper/DataTableHelper.php [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA ;
+ *
+ */
+
```
Comment (gabrielfs7) [inline helper/DataTableHelper.php:21]:
```suggestion
declare(strict_types=1);
```

File: model/TestCenterFormService.php [hunk 6]
```diff
@@ -28,7 +30,7 @@
 
 class TestCenterFormService extends ConfigurableService
 {
-    const SERVICE_ID = 'taoTestCenter/TestCenterFormService';
+    const SERVICE_ID = 'taoTestCenterRostering/TestCenterFormService';
```
Comment (gabrielfs7) [inline model/TestCenterFormService.php]:
```suggestion
    public const SERVICE_ID = 'taoTestCenterRostering/TestCenterFormService';
```

File: model/import/TestCenterAdminCsvImporter.php [hunk 7]
```diff
@@ -57,8 +58,9 @@ class TestCenterAdminCsvImporter extends RdsUserImportService
     public function import($file, $extraProperties = [], $options = [])
     {
         $extraProperties[UserRdf::PROPERTY_ROLES] = TestCenterService::ROLE_TESTCENTER_ADMINISTRATOR;
-        $extraProperties['roles'] = TestCenterService::ROLE_TESTCENTER_ADMINISTRATOR;;
+        $extraProperties['roles'] = TestCenterService::ROLE_TESTCENTER_ADMINISTRATOR;
+        ;
```
Comment (gabrielfs7) [inline model/import/TestCenterAdminCsvImporter.php]: - [ ] Remove extra `;`

**Review by gabrielfs7 — COMMENTED** [review]
I did some comments about optimisation like, constant visibility, missing type hints and return type, static methods etc, but not sure they apply to this PR, so ignore if not.

**Review by gabrielfs7 — APPROVED** [review]
Good work! LGTM. As it is huge, let's ask for QA to test in proper env.

**Pull request merged**